Set up dokka and docs publishing process (#780)
Kotlin/kotlinx.serialization: dokka/moduledoc.md

# Module kotlinx-serialization-runtime

> A runtime library that makes usage of code generated by kotlinx.serialization compiler plugin.

Reviewer: ```suggestion
A companion library to kotlinx.serialization compiler plugin.
```

Contains core serialization API, as well as serializers for standard library classes, and ready for use JSON

> format implementation.

Reviewer: ```suggestion
Contains core serialization API, serializers for standard library classes, and ready to use JSON
format implementation.
```
